src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3211,3 +3224,236 @@ def _aks_approuting_update(
+        if not instances:
+            raise ValidationError(f"No instances found in the VMSS {vmss_name}!")
+
+        instance_id = instances[0].instance_id
+        logger.debug("Select instance id: %s", instance_id)
+    else:
+        index = node_name.find("vmss")
+        if index != -1:
+            vmss_name = node_name[:index + 4]
+            instance_id = int(node_name[index + 4:], base=36)
+            instance_info = compute_client.virtual_machine_scale_set_vms.get(
+                managed_resource_group, vmss_name, instance_id)
+            if not instance_info:
+                raise ValidationError(f"Instance id {instance_id} not found in VMSS {vmss_name}!")
+            _aks_verify_resource(instance_info, CONST_VIRTUAL_MACHINES)
+        else:
+            raise ValidationError(f"Node name {node_name} is invalid!")
+
+    return vmss_name, instance_id
+
+
+def _aks_get_node_name_as(
+        cmd,
+        node_name,
+        managed_resource_group):
+    compute_client = get_compute_client(cmd.cli_ctx)
+
+    if not node_name:
+        print("No node name specified, will randomly select a node from the cluster")
+
+        vm_list = compute_client.virtual_machines.list(managed_resource_group)
+        if not vm_list:
+            raise ValidationError(f"No VM found in the managed resource group {managed_resource_group}!")
+
+        vm_name = ""
+        for vm in vm_list:
+            try:
+                _aks_verify_resource(vm, CONST_VIRTUAL_MACHINES)
+                vm_name = vm.name
+                logger.debug("Select VM: %s", vm_name)
+                break
+            except ValidationError as ex:
+                logger.warning(ex)
+                continue
+
+        if not vm_name:
+            raise ValidationError("No suitable VM found in the managed resource!")
+    else:
+        vm_name = node_name
+        vm_info = compute_client.virtual_machines.get(managed_resource_group, vm_name)
+        if not vm_info:
+            raise ValidationError(f"VM {vm_name} not found in the managed resource group {managed_resource_group}!")
+        _aks_verify_resource(vm_info, CONST_VIRTUAL_MACHINES)
+
+    return vm_name
+
+
+def aks_check_network_outbound(
+        cmd,
+        client,
+        resource_group_name,
+        cluster_name,
+        node_name=None,
+        custom_endpoints=None):
+    fqdn = ""
+    cluster = aks_show(cmd, client, resource_group_name, cluster_name, None)
+    if not cluster:
+        logger.warning("Can not get cluster information!")
+    else:
+        fqdn = cluster.fqdn
+        if not fqdn:
+            logger.warning("Can not get cluster api server!")
+        else:
+            print("Get cluster api server:", fqdn)
+
+        vm_set_type = cluster.agent_pool_profiles[0].type
+        if not vm_set_type:
+            raise ValidationError("Can not get VM set type of the cluster!")
+        print("Get node pool VM set type:", vm_set_type)
```

Comment:
consider logger.debug

Reply:
for this, I do want it to be part of the stdout